pkg/3scale/amp/component/system_monitoring.go

```diff
@@ -30,6 +30,75 @@ func (system *System) SystemSidekiqPodMonitor() *monitoringv1.PodMonitor {
 	}
 }
 
+func (system *System) SystemAppPodMonitor() *monitoringv1.PodMonitor {
+	return &monitoringv1.PodMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "system-app",
+			Labels: system.Options.CommonSidekiqLabels,
+		},
+		Spec: monitoringv1.PodMonitorSpec{
+			PodMetricsEndpoints: []monitoringv1.PodMetricsEndpoint{
+				{
+					Port:   "masterMetrics",
+					Path:   "/metrics",
+					Scheme: "http",
+				},
+				{
+					Port:   "providerMetrics",
+					Path:   "/metrics",
+					Scheme: "http",
+				},
+				{
+					Port:   "developerMetrics",
+					Path:   "/metrics",
+					Scheme: "http",
+				},
+			},
+			Selector: metav1.LabelSelector{
+				MatchLabels: system.Options.CommonAppLabels,
+			},
+		},
+	}
+}
+
+func (system *System) SystemAppProviderContainerPodMonitor() *monitoringv1.PodMonitor {
+	return &monitoringv1.PodMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "system-app-master-container",
+			Labels: system.Options.CommonSidekiqLabels,
+		},
+		Spec: monitoringv1.PodMonitorSpec{
+			PodMetricsEndpoints: []monitoringv1.PodMetricsEndpoint{{
+				Port:   "masterMetrics",
```

Comment:
Notice how here the port is 'masterMetrics' and not just 'metrics'. In system component I've assigned different `ContainerPort` names for each container in system-app. The reason for that is that it seems that PodMonitor is at PodLevel and the way to differentiate between endpoints in the pod is by the Port name or by path or by scheme.
So, to differentiate between the container ports in system-app I've had to set unique names for them.

What do you think?